Attendance/Projects/Shared/DesignSystem/Sources/UI/CustomDate/View/CustomFSCalendarView.swift

```diff
@@ -0,0 +1,158 @@
+  
+  public init(
+    selectDate: Binding<Date>,
+    currentMonth: Binding<Date>
+  ) {
+    self._selectDate = selectDate
+    self._currentMonth = currentMonth
+  }
+  
+  public func makeCoordinator() -> Coordinator {
+    Coordinator(parent: self)
+  }
+  
+  // MARK: - ui 그리기
+  public func makeUIView(context: Context) -> UIView {
+    let containerView = UIView()
+    
+    // FSCalendar 인스턴스 생성 및 설정
+    let calendar = FSCalendar()
+    calendar.delegate = context.coordinator
+    calendar.dataSource = context.coordinator
+    context.coordinator.calendar = calendar
+    calendar.scrollDirection = .horizontal
+    calendar.scrollEnabled = true
+    calendar.allowsMultipleSelection = false
+    calendar.appearance.selectionColor = .primaryBlue
+    calendar.appearance.titleSelectionColor = .white
+    
+    // 외관 설정
+    calendar.appearance.headerDateFormat = " "
+    calendar.appearance.weekdayTextColor = .black
+    calendar.appearance.headerTitleAlignment = .center
+    calendar.appearance.borderRadius = 20
+    calendar.appearance.todayColor = .white
+    calendar.appearance.titleTodayColor = .black
+    calendar.placeholderType = .none
+    calendar.swipeToChooseGesture.isEnabled = false
+    calendar.locale = Locale(identifier: "ko_KR")
+    calendar.appearance.headerMinimumDissolvedAlpha = 0.0
+    calendar.headerHeight = 20
+    calendar.backgroundColor = .white
+    
+    calendar.layer.cornerRadius = 20
+    calendar.layer.masksToBounds = true
+    
+    containerView.addSubview(calendar)
+    calendar.translatesAutoresizingMaskIntoConstraints = false
+    
+    NSLayoutConstraint.activate([
+      calendar.topAnchor.constraint(equalTo: containerView.topAnchor),
+      calendar.leadingAnchor.constraint(equalTo: containerView.leadingAnchor, constant: 20),
+      calendar.trailingAnchor.constraint(equalTo: containerView.trailingAnchor, constant: -20),
+      calendar.bottomAnchor.constraint(equalTo: containerView.bottomAnchor)
+    ])
+    
+    // 오늘 날짜 선택하지 않고, 현재 페이지만 설정
+    DispatchQueue.main.async {
+      context.coordinator.calendar.setCurrentPage(selectDate, animated: true)
+      context.coordinator.calendar.reloadData()
+    }
+    
+    return containerView
+  }
+  
+  public func updateUIView(_ uiView: UIView, context: Context) {
+    if let calendar = context.coordinator.calendar {
+      calendar.reloadData()
+      // 자동 업데이트된 날짜에도 텍스트 컬러 적용을 위해 다시 선택 상태로 만듭니다.
+      calendar.select(selectDate)
+      updateHeaderTitle(coordinator: context.coordinator)
+    }
+  }
+  
+  // 헤더 제목 업데이트 메서드
+  public func updateHeaderTitle(coordinator: Coordinator) {
+    guard let calendar = coordinator.calendar else { return }
+    DispatchQueue.main.async {
+      calendar.currentPage =  coordinator.parent.currentMonth
+    }
+  }
+  
+  public class Coordinator: NSObject, FSCalendarDelegate, FSCalendarDataSource, FSCalendarDelegateAppearance {
+    let parent: CustomFSCalendarView
+    var calendar: FSCalendar!
```

Comment:
```suggestion
    weak var calendar: FSCalendar?
```
순환 참조 방지랑 보다 안전하게 weak var로 수정하는게 더 좋아보여요~